### PR TITLE
[5.4] Add App\User to Guard::user() possible return values

### PR DESCRIPTION
**Pull request by KKSzymanowski**
I personally almost always stick with the default App\User model and this provides autocompletion for custom methods defined on it in IDEs like PhpStorm. 

image

For those who choose to change the default location of User model it will make no difference.

### CODE REVIEW

File: src/Illuminate/Contracts/Auth/Guard.php [hunk 1]
```diff
@@ -21,7 +21,7 @@ public function guest();
     /**
      * Get the currently authenticated user.
      *
-     * @return \Illuminate\Contracts\Auth\Authenticatable|null
+     * @return \Illuminate\Contracts\Auth\Authenticatable|\App\User|null
```
Comment (laurencei) [inline src/Illuminate/Contracts/Auth/Guard.php]: You cant hardcode `\App\User` here.

Firstly - what happens if you have your User class elsewhere (I dont keep mine there).

Secondly - not everyone uses the `App` namespace.

Comment (KKSzymanowski) [inline src/Illuminate/Contracts/Auth/Guard.php]: As I said, this change won't make any difference if the User model is somewhere else. IDE will not find the class and provide completion from only the Authenticatable contract.

It has no side effects as far as I tested(have you encountered any?) and it makes life a little bit easier for those who stick to the defaults.

Comment (laurencei) [inline src/Illuminate/Contracts/Auth/Guard.php]: If you use a `App\User` class - but use a different class for Authenticating - that scenario would give the wrong autocomplete.

Anyway - regardless - its not about side effects - its about hard coding an App variable into a framework that should not have knowledge of it. That is why the contracts exists.

Comment (KKSzymanowski) [inline src/Illuminate/Contracts/Auth/Guard.php]: Yes, you have a valid point. 
I just thought it would be convenient for most users.

**Comment by taylorotwell** [comment]
It doesn't make a lot of sense to put \App\User in the framework.

**Pull request closed without merging**